packages/core/src/setup.ts

```diff
@@ -216,7 +216,8 @@ export function setup<
     TInput,
     TOutput,
     TEmitted,
-    TMeta
+    TMeta,
+    TConfig
```

Comment:
@Andarist I included `TConfig` because providing just `ToStateValue<TConfig>` loses too much information. I think I prefer passing the `TConfig` (as a `StateSchema`) instead of reducing the types in a lossy way. WDYT?

Reply:
Conceptually it would be better to pass minimal required information down here instead of the whole `TConfig`. That said, with this change, it becomes even less likely for one to create 2 different state machines that could be used interchangeably... so 🤷‍♂️